lib/rage/controller/api.rb

```diff
@@ -283,4 +288,42 @@ def headers
     @__headers = {}.merge!(@__headers) if DEFAULT_HEADERS.equal?(@__headers)
     @__headers
   end
+
+  class Request
+    SPECIAL_HEADERS = %w[CONTENT_TYPE CONTENT_LENGTH].freeze
+    # Get the request headers.
+    # @example
+    #  request.headers["Content-Type"] # => "application/json"
+    # or request.headers["HTTP_CONTENT_TYPE"] # => "application/json"
+    attr_reader :headers
+
+    def initialize(env)
+      @env = env
+    end
+
+    def headers
+      @headers ||= Headers.new(@env)
+    end
+
+    class Headers
+      def initialize(env)
+        @env = env
+        @headers = {}
+      end
+
+      def [](requested_header)
+        requested_header = requested_header.upcase.tr('-', '_')
+
+        if requested_header.start_with?("HTTP_")
+          @env[requested_header]
+        else
+          normalized_name = "HTTP_" + requested_header
+          if SPECIAL_HEADERS.include?(requested_header)
+            normalized_name = requested_header
+          end
+          @env[normalized_name]
```

Comment:
again, for performance reasons, let's update this:

```suggestion
          if "CONTENT_TYPE" == requested_header || "CONTENT_LENGTH" == requested_header
            @env[requested_header]
          else
            (requested_header = requested_header.tr('-', '_')).upcase!
            @env["HTTP_#{requested_header}"]
          end
```